### PR TITLE
[MOD] smolraptor borgs

### PR DESCRIPTION
**Pull request by aKromatopzia**
## About The Pull Request

smaller raptor borgs. i might bring these to skyrat but these will be here for now. Ain't doing more before people say if they're fine with the art style or not.

has a sprite for https://github.com/Bubberstation/Bubberstation/pull/1407

## Proof Of Testing

https://github.com/Bubberstation/Bubberstation/assets/94389683/9c5148e7-dfc7-4355-89ee-6dfa10c01a6c

Screenshot_90
Screenshot_91

above images are minorly outdated.

## Changelog

:cl:
add: new silicon model (smolraptor). non-syndie, sorry; for now
/:cl:

### CODE REVIEW

**Comment by Erol509** [comment]
They loook awesome

**Comment by NargaInDisguise** [comment]
co-starring: my voice lmao????
commenting here before it remakes the video without me in it

**Comment by aKromatopzia** [comment]
Update: CC model and some colours got shifted.

**Comment by aKromatopzia** [comment]
bugfix and sprite changes, ready again

**Comment by Superhats** [comment]
image
I don't know how this happened but I joined the round and my sprite was in between two tiles

**Comment by aKromatopzia** [comment]
> image I don't know how this happened but I joined the round and my sprite was in between two tiles

this tends to happen with two-wide borgs; find your pixel shift button and move along the misaligned axis until you can't any more. then move. it should reset you.

**Comment by Superhats** [comment]
> > image I don't know how this happened but I joined the round and my sprite was in between two tiles
> 
> this tends to happen with two-wide borgs; find your pixel shift button and move along the misaligned axis until you can't any more. then move. it should reset you.

I tried doing that, it did not fix it

**Comment by thegrb93** [comment]
> image I don't know how this happened but I joined the round and my sprite was in between two tiles

yeah It's currently happening

**Comment by aKromatopzia** [comment]
TRAIT_R_WIDE has been given back to them, and they now have appropriate emote sprites.

**Comment by Superhats** [comment]
Pretty significant problem. The Raptorborg's panel isn't visible from the front, so there's no feedback to indicate it's open. I've personally gotten emagged twice because someone opened my panel while I was talking to them and there was no prompt or other visual to warn me.

**Comment by thegrb93** [comment]
> Pretty significant problem. The Raptorborg's panel isn't visible from the front, so there's no feedback to indicate it's open. I've personally gotten emagged twice because someone opened my panel while I was talking to them and there was no prompt or other visual to warn me.

Thats true for any borg, you just gotta watch for them grabbing their emag and crowbar. If your hatch is open and you haven't already noticed, its already too late.

I think the raptor's hatch is on the back too. I've seen it open.

**Comment by aKromatopzia** [comment]
> Pretty significant problem. The Raptorborg's panel isn't visible from the front, so there's no feedback to indicate it's open. I've personally gotten emagged twice because someone opened my panel while I was talking to them and there was no prompt or other visual to warn me.

other borgs have their panel visible from the front and sides; raptors are on the back and sides. It is what it is.

**Comment by projectkepler-ru** [comment]
science variant perhaps

**Comment by ReturnToZender** [comment]
> science variant perhaps

Sprites for the science variant are here but until research borgs get merged they can't be put in :(

File: modular_zubbers/modules/smolraptors/code/smallraptor.dm [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+#define CYBORG_ICON_GEN_SMOLRAPTOR 'modular_zubbers/modules/smolraptors/icons/smolraptor_gen.dmi'
+#define CYBORG_ICON_SCI_SMOLRAPTOR 'modular_zubbers/modules/smolraptors/icons/smolraptor_sci.dmi'
+#define CYBORG_ICON_ENG_SMOLRAPTOR 'modular_zubbers/modules/smolraptors/icons/smolraptor_eng.dmi'
+#define CYBORG_ICON_MED_SMOLRAPTOR 'modular_zubbers/modules/smolraptors/icons/smolraptor_med.dmi'
+#define CYBORG_ICON_CAR_SMOLRAPTOR 'modular_zubbers/modules/smolraptors/icons/smolraptor_car.dmi'
+#define CYBORG_ICON_SERV_SMOLRAPTOR 'modular_zubbers/modules/smolraptors/icons/smolraptor_serv.dmi'
+#define CYBORG_ICON_PK_SMOLRAPTOR 'modular_zubbers/modules/smolraptors/icons/smolraptor_pk.dmi'
+#define CYBORG_ICON_JANI_SMOLRAPTOR 'modular_zubbers/modules/smolraptors/icons/smolraptor_jani.dmi'
+#define CYBORG_ICON_MIN_SMOLRAPTOR 'modular_zubbers/modules/smolraptors/icons/smolraptor_min.dmi'
+#define CYBORG_ICON_CC_SMOLRAPTOR 'modular_zubbers/modules/smolraptors/icons/smolraptor_cc.dmi'
```
Comment (Majkl-J) [inline modular_zubbers/modules/smolraptors/code/smallraptor.dm:16]: Add undefs on the end of the file.

**Review by Majkl-J — CHANGES_REQUESTED** [review]
Looks good to me otherwise, once you add the undefs this should be ready to merge.

File: modular_zubbers/modules/smolraptors/code/smallraptor.dm [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+/obj/item/robot_model/ninja/Initialize(mapload)
+	. = ..()
+	borg_skins |= list(
+		"SmolRaptor" = list(SKIN_ICON_STATE = "smolraptor", SKIN_ICON = CYBORG_ICON_NINJ_SMOLRAPTOR, SKIN_FEATURES = list(TRAIT_R_UNIQUEWRECK, TRAIT_R_SMALL, TRAIT_R_WIDE)),
+	)
+
+*/
+
+#undef CYBORG_ICON_TYPE_SMOLRAPTOR "smolraptor"
```
Comment (projectkepler-ru) [inline modular_zubbers/modules/smolraptors/code/smallraptor.dm:104]:
```suggestion
#undef CYBORG_ICON_TYPE_SMOLRAPTOR 
```

**Comment by CenkTheRaptor** [comment]
YEAAAAAAH BABYYYYYY THATS WHAT IVE BEEN WAITING FOR THATS WHAT ITS ALL ABOUT WOOOOOOOOOOOOOOOOOOOOOOOOO

**Comment by Erol509** [comment]
What about skin hat offset?

**Comment by thegrb93** [comment]
> What about skin hat offset?

Is this what controls hat position? I know the other wide borgs have hat wearing issues.

**Comment by Erol509** [comment]
> > What about skin hat offset?
> 
> Is this what controls hat position? I know the other wide borgs have hat wearing issues.

Yes, it shifts the hat offset so it can land on the  actual head. its a pixel shift

**Comment by thegrb93** [comment]
> Yes, it shifts the hat offset so it can land on the actual head. its a pixel shift

Even if the borg's head position changes depending on rotation? I think the problem with wide borgs is their head is not stationary. That would be cool if it works for any rotation though. I would fix some of them if it does have the capability.

**Review by Majkl-J — APPROVED** [review]

**Comment by pixelkitty286** [comment]
oh god i should of told them to fix file paths now i gotta clean that AAAAAAA

**Pull request merged**